sympy/combinatorics/pc_groups.py

```diff
@@ -0,0 +1,201 @@
+from sympy.combinatorics.free_groups import free_group, FreeGroupElement
+from sympy.combinatorics.fp_groups import FpGroup, FpSubgroup, subgroup_quotient
+from sympy import sieve
+
+class PcGroup(FpGroup):
+    '''
+    Class implementing Polycyclic groups.
+    To-do : Docstring completion.
+            Add error and exception handling
+    The generators will be converted to equivalent forms on
+    another free group.
+    The implementation for this methods will be similar to
+    the coin change problem in DP.
+    '''
+    def __init__(self, pc_series, power_exponents=None):
```

Comment:
A user should be submitting either `pc_series` or `power_exponents`, not both at the same time. It makes more sense to take some `arg`, work out if it specifies `pc_series` or `power_exponents`, then proceed based on that. I still think you should implement pc presentation from `power_exponents` as soon as possible because then you could be testing all polycyclic methods you are adding. Computing polycyclic series is hard and will take time - I'm actually wondering if it should be a separate PR. But computing a presentation given power exponents is easy